sql/src/test/java/io/crate/operation/collect/files/CSVLineParserTest.java

```diff
@@ -0,0 +1,155 @@
+        givenRow("GER,Germany\n");
+
+        whenParseIsCalled();
+    }
+
+    @Test
+    public void parse_givenCSVInput_thenParsesToJson() throws IOException {
+        givenHeader("Code,Country\n");
+        givenRow("GER,Germany\n");
+
+        whenParseIsCalled();
+
+        thenResultIs("{\"Code\":\"GER\",\"Country\":\"Germany\"}");
+    }
+
+    @Test
+    public void parse_givenEmptyRow_thenParsesToEmptyJson() throws IOException {
+        givenHeader("Code,Country\n");
+        givenRow("\n");
+
+        whenParseIsCalled();
+
+        thenResultIs("{}");
+    }
+
+    @Test
+    public void parse_givenEscapedComma_thenParsesLineCorrectly() throws IOException {
+        givenHeader("Code,\"Coun, try\"\n");
+        givenRow("GER,Germany\n");
+
+        whenParseIsCalled();
+
+        thenResultIs("{\"Code\":\"GER\",\"Coun, try\":\"Germany\"}");
+    }
+
+
+    @Test
+    public void parse_givenRowWithMissingValue_thenTheValueIsAssignedToKeyAsAnEmptyString() throws IOException {
+        givenHeader("Code,Country,City\n");
+        givenRow("GER,,Berlin\n");
+
+        whenParseIsCalled();
+
+        thenResultIs("{\"Code\":\"GER\",\"Country\":\"\",\"City\":\"Berlin\"}");
+    }
+
+
+    @Test
+    public void parse_givenTrailingWhiteSpaceInHeader_thenParsesToJsonWithoutWhitespace() throws IOException {
+        givenHeader("Code ,Country  \n");
+        givenRow("GER,Germany\n");
+
+        whenParseIsCalled();
+
+        thenResultIs("{\"Code\":\"GER\",\"Country\":\"Germany\"}");
+    }
+
+    @Test
+    public void parse_givenTrailingWhiteSpaceInRow_thenParsesToJsonWithoutWhitespace() throws IOException {
+        givenHeader("Code,Country\n");
+        givenRow("GER        ,Germany\n");
+
+        whenParseIsCalled();
+
+        thenResultIs("{\"Code\":\"GER\",\"Country\":\"Germany\"}");
+    }
+
+    @Test
+    public void parse_givenPrecedingWhiteSpaceInHeader_thenParsesToJsonWithoutWhitespace() throws IOException {
+        givenHeader("         Code,         Country\n");
+        givenRow("GER,Germany\n");
+
+        whenParseIsCalled();
+
+        thenResultIs("{\"Code\":\"GER\",\"Country\":\"Germany\"}");
+    }
+
+    @Test
+    public void parse_givenPrecedingWhiteSpaceInRow_thenParsesToJsonWithoutWhitespace() throws IOException {
+        givenHeader("Code,Country\n");
+        givenRow("GER,               Germany\n");
+
+        whenParseIsCalled();
+
+        thenResultIs("{\"Code\":\"GER\",\"Country\":\"Germany\"}");
+    }
+
+    private void givenHeader(String header) {
+        headerByteArray = header.getBytes(StandardCharsets.UTF_8);
+    }
+
+    private void givenRow(String row) {
+        rowByteArray = row.getBytes(StandardCharsets.UTF_8);
+    }
+
+    private void whenParseIsCalled() throws IOException {
+        result = subjectUnderTest.parse(headerByteArray, rowByteArray);
+    }
+
+    private void thenResultIs(String expected) throws IOException {
+        assertThat(result, is(expected));
```

Comment:
I love how you name the test methods. However, I feel like you're overdoing it for some of the test methods. I find `thenResultIs("result")` not easier to read than `assertThat(result, is(expected));` because I'm not sure which result is being checked in the former case. Not really a blocker, just something to consider :)

Reply:
I'd also appreciate it if we could stick to the `assertThat.. is` pattern. No matter what may or may not be more readable, I think we should keep things consistent.

Reply:
I was going by (gherkin syntax)[https://github.com/cucumber/cucumber/wiki/Gherkin]  for these tests, hence the given, when, then steps.

I've reverted the `assert` steps to `assertThat.. is`, though imo it does slightly reduce readability because there is more noise. 
Also, while consistency is important, I don't think readability of code should be dismissed or undervalued. Imo a well-written set of tests does two things: 1. (Obviously) test the code and edge cases 2. Act as a form of living documentation for outlining what the code it's testing does. You should be able to glance at a set of tests and immediately understand what they're doing, and how the class it's testing works, and imo this style of testing really facilitates that. :)

Reply:
I think there is a trade-off at some point. If you create your own testing DSL, readers will have to learn it first (which defeats the purpose of readability then). Regular assertions and local variables can be consumed much easier than custom assertion code and class-level variables.

Reply:
I agree with Max, if we start creating this kind of syntax for all unit tests that means each time the readers have to understand the DSL implementation *and* the tests. All this indirection and global variable usage is too much for simple unit tests. 

> while consistency is important, I don't think readability of code should be dismissed

Consistency = readability to a certain degree because people get used to reading the similar patterns.